src/main/java/edu/cornell/kfs/fp/document/RecurringDisbursementVoucherDocument.java

```diff
@@ -0,0 +1,86 @@
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntry;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySequenceHelper;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySourceDetail;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.coreservice.framework.parameter.ParameterConstants.COMPONENT;
+import org.kuali.rice.coreservice.framework.parameter.ParameterConstants.NAMESPACE;
+import org.kuali.rice.kew.api.exception.WorkflowException;
+import org.kuali.rice.kew.framework.postprocessor.DocumentRouteStatusChange;
+
+import edu.cornell.kfs.fp.businessobject.RecurringDisbursementVoucherDetail;
+import edu.cornell.kfs.fp.service.RecurringDisbursementVoucherDocumentService;
+
+@NAMESPACE(namespace = KFSConstants.CoreModuleNamespaces.FINANCIAL)
+@COMPONENT(component = "RecurringDisbursementVoucher")
+public class RecurringDisbursementVoucherDocument extends CuDisbursementVoucherDocument {
+
+	private static final long serialVersionUID = 5578411133159987973L;
+	protected static Log LOG = LogFactory.getLog(RecurringDisbursementVoucherDocument.class);
+	
+	private List<RecurringDisbursementVoucherDetail> recurringDisbursementVoucherDetails;
+	
+	@Override
+    public void customizeExplicitGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySourceDetail postable, GeneralLedgerPendingEntry explicitEntry) {
+		//Intentionally left blank as we don't need to create GLPE from a recurring DV
+    }
+	
+	@Override
+    protected void processExplicitGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySequenceHelper sequenceHelper,
+        GeneralLedgerPendingEntrySourceDetail generalLedgerPendingEntryDetail, GeneralLedgerPendingEntry explicitEntry) {
+		//Intentionally left blank as we don't need to create GLPE from a recurring DV
+    }
+	
+	
+	@Override
+    public boolean generateGeneralLedgerPendingEntries(GeneralLedgerPendingEntrySourceDetail glpeSourceDetail,
+        GeneralLedgerPendingEntrySequenceHelper sequenceHelper) {
+		//Intentionally left blank as we don't need to create GLPE from a recurring DV
+        return true;
+    }
+	
+	@Override
+    protected boolean processOffsetGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySequenceHelper sequenceHelper,
+        GeneralLedgerPendingEntrySourceDetail postable, GeneralLedgerPendingEntry explicitEntry, GeneralLedgerPendingEntry offsetEntry) {
+		//Intentionally left blank as we don't need to create GLPE from a recurring DV
+		return true;
+    }
+	
+	@Override
+    public void doRouteStatusChange(DocumentRouteStatusChange statusChangeEvent) {
+		LOG.info("called postProcessSave");
+		super.doRouteStatusChange(statusChangeEvent);
+		if (getDocumentHeader().getWorkflowDocument().isProcessed()) {
+        	try {
+        		getRecurringDisbursementVoucherDocumentService().generateDisbursementDocumentsFromRecurringDV(this);
+			} catch (WorkflowException e) {
+				LOG.error(e);
```

Comment:
would be nice to add context to this log message and the exception we're rethrowing